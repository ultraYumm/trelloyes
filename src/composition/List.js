import React from 'react'
import './List.css';
import Card from './Card'
/*import './src/STORE.js';*/



export default function List(props) {
  
  return (
    <section className='List'>
      <header className='List-header'>
        <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {props.cards.map((card) =>
          <Card
            key={card.id}
            id={card.id}
            title={card.title}
            content={card.content}
            onClickDelete={props.onClickDelete}
          />
        )}
        <button
          onClick={() => props.onClickAdd(props.id)}
          type='button'
          className='List-add-button'
        >
          + Add Random Card
        </button>
      </div>
    </section>
  )
}

List.defaultProps = {
  onClickAdd: () => {},
}