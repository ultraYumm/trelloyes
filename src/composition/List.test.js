import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const section = document.createElement('section');
    ReactDOM.render(<List />, section);
  
    // clean up code
    ReactDOM.unmountComponentAtNode(section);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List
        key={list.id}
        header={list.header}
        cards={list.cardIds.map(id => store.allCards[id])}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });