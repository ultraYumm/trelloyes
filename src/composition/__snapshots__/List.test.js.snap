// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        my title
      </h3>
      <p>
        some content
      </p>
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
